### PR TITLE
feat: timeline

### PR DESCRIPTION
**Pull request by paul-nguyen-goldenowl**
## Description
- [x] Add message timeline 
- [x] Fix message timestamp

### CODE REVIEW

File: simplechat/src/main/java/com/ln/simplechat/ui/chat/ChatAdapter.kt [hunk 1]
```diff
@@ -89,7 +110,11 @@ class ChatAdapter(
                     loadImages(messageImageRcv, outGoing, item.medias)
                 }
             }
-            timestamp.text = dateFormatter.format(item.timestamp)
+            timestamp.text = DateUtils.SDF_HOUR.format(item.timestamp)
+            /* if next message is from the same sender, hide timestamp, show on item long clicked */
+            var nextMessengerSame = nextItem?.sender == item.sender
+            timestamp.visibility = if (nextMessengerSame && sameDateAfter) View.GONE else View.VISIBLE
```
Comment (dustin-goldenowl) [inline simplechat/src/main/java/com/ln/simplechat/ui/chat/ChatAdapter.kt]: Can refer this `timestamp.isVisible = !(nextMessengerSame && sameDateAfter)`

File: simplechat/src/main/java/com/ln/simplechat/utils/DateUtils.kt [hunk 2]
```diff
@@ -58,6 +62,28 @@ object DateUtils {
...
+        val paramDay: Int = calendar.get(Calendar.DAY_OF_YEAR)
+        return currentDay - paramDay == 1
+    }
+
+    fun isToday(time: Long): Boolean {
+        return DateUtils.isToday(time)
+    }
+
+    fun isSameDate(date1: Long, date2: Long): Boolean {
+        val cal1 = Calendar.getInstance();
```
Comment (dustin-goldenowl) [inline simplechat/src/main/java/com/ln/simplechat/utils/DateUtils.kt]: let rename meaning

File: simplechat/src/main/java/com/ln/simplechat/utils/DateUtils.kt [hunk 3]
```diff
@@ -58,6 +62,28 @@ object DateUtils {
...
+        val cal2 = Calendar.getInstance();
+        cal2.time = Date(date2)
+        return (cal1.get(Calendar.ERA) == cal2.get(Calendar.ERA) &&
+                cal1.get(Calendar.YEAR) == cal2.get(Calendar.YEAR) &&
+                cal1.get(Calendar.DAY_OF_YEAR) == cal2.get(Calendar.DAY_OF_YEAR))
+    }
+
     fun dateDiffer(d: Long): Int {
         return try {
             val l1 = currentTimeMillis
```
Comment (dustin-goldenowl) [inline simplechat/src/main/java/com/ln/simplechat/utils/DateUtils.kt:89]: let rename meaning

Comment (paul-nguyen-goldenowl) [reply simplechat/src/main/java/com/ln/simplechat/utils/DateUtils.kt:89]: renamed

**Pull request merged**